lecture_2/homework/app/models/hussar.rb

```diff
@@ -0,0 +1,7 @@
+class Hussar < Warrior
+  has_one :weapon, class_name: "Magical", foreign_key: "warrior_id", dependent: :nullify
```

Comment:
Widzę, że bardzo restrykcyjne podszedłeś do informacji, że każda jednostka może używać tylko jednego typu broni 😃 Spróbuj zastanowić się czy jest jakiś sposób, żeby nie musieć tej relacji powtarzać w każdym modelu. Najlepiej byłoby gdybyśmy zdefiniowali ją w klasie `Warrior` a w konkretnych typach wojowników mogli po prostu wybrać który typ jest akceptowany

Reply:
No i jest z tym jeszcze inna zagadka. Jeśli stworzymy broń `Melee` i przypiszemy ją do husarza to aplikacja nie będzie miała żadnych zastrzeżeń. Gdzie i jak możemy rozwiązać ten problem?

Reply:
Hm, w takim razie rzeczywiście źle zrozumiałem ten podpunkt i chyba za bardzo nie mam pomysłu jak to zrobić, ale jeszcze pokombinuję z polimorfizmem i postaram się wrzucić :P